### ISSUE TITLE
Migration Error: Executing erpnext.patches.v5_0.update_item_description_and_image

### ISSUE THREAD

**Issue by revant** [frappe/erpnext#3057]
bench switch-to-master:
```
revant@revant-laptop:~/frappe-bench$ bench switch-to-master
Switched to branch 'master'
Your branch is up-to-date with 'upstream/master'.
Switched to branch 'master'
Your branch is up-to-date with 'upstream/master'.
Switched to branch 'master'
Your branch is up-to-date with 'upstream/master'.

Switched to master
Please run `bench update --patch` to be safe from any differences in database schema
```
Install new site:
```
revant@revant-laptop:~/frappe-bench$ bench new-site site1.local
MySQL root password: 
Created user 1bd3e0294d
Created database 1bd3e0294d
Granted privileges to user 1bd3e0294d and database 1bd3e0294d
Starting database import...
Imported from database /home/revant/frappe-bench/apps/frappe/frappe/data/Framework.sql
Updating frappe                     : [========================================]
Set Administrator password: 
Re-enter Administrator password: 
Installing Fixtures...
*** Scheduler is disabled ***
```
Install erpnext and shopping_cart on site:
```
revant@revant-laptop:~/frappe-bench$ bench frappe --install_app erpnext site1.local
Updating erpnext                    : [========================================]
Importing Chart of Accounts
Installing Fixtures...
revant@revant-laptop:~/frappe-bench$ bench frappe --install_app shopping_cart site1.local
Updating shopping_cart              : [========================================]
Installing Fixtures...
```
Start v4 site and fill in mandatory details in setup:
```
revant@revant-laptop:~/frappe-bench$ bench set-default-site site1.local
revant@revant-laptop:~/frappe-bench$ bench setup nginx
revant@revant-laptop:~/frappe-bench$ bench setup supervisor
revant@revant-laptop:~/frappe-bench$ sudo service nginx reload
revant@revant-laptop:~/frappe-bench$ sudo service supervisor restart

```
Switch to develop and check update requirements:
```
revant@revant-laptop:~/frappe-bench$ bench switch-to-develop
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.

Switched to develop
Please run `bench update --patch` to be safe from any differences in database schema
revant@revant-laptop:~/frappe-bench$ bench update --requirements
frappe-web: stopped
frappe-workerbeat: stopped
frappe-worker: stopped
frappe-web: started
frappe-workerbeat: started
frappe-worker: started
________________________________________________________________________________
https://frappe.io/buy - Donate to help make better free and open source tools
```
Migrate to v5:
```
revant@revant-laptop:~/frappe-bench$ bench update --upgrade
Already up-to-date.
From https://github.com/frappe/shopping-cart
 * branch            develop    -> FETCH_HEAD
Already up-to-date.
From https://github.com/frappe/frappe
 * branch            develop    -> FETCH_HEAD
Already up-to-date.
From https://github.com/frappe/erpnext
 * branch            develop    -> FETCH_HEAD
Already up-to-date.
Migrating site1.local
Executing frappe.patches.v5_0.convert_to_barracuda_and_utf8mb4 in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('core', 'doctype', 'docfield', force=True) #2015-01-01 in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('email', 'doctype', 'email_alert') #2014-07-15 in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('desk', 'doctype', 'todo') #2014-12-31-1 in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('custom', 'doctype', 'property_setter') #2014-12-31-1 in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.v4_to_v5 in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.remove_shopping_cart_app in site1.local (1bd3e0294d)
Success
Executing execute:frappe.permissions.reset_perms("Module Def") in site1.local (1bd3e0294d)
Success
Executing execute:frappe.permissions.reset_perms("User") #2015-03-24 in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.rename_table_fieldnames in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.communication_parent in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.clear_website_group_and_notifications in site1.local (1bd3e0294d)
Success
Executing execute:frappe.db.sql("""update tabComment set comment = substr(comment, 6, locate(":", comment)-6) where comment_type in ("Assigned", "Assignment Completed")""") in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.fix_feed in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.update_shared in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.bookmarks_to_stars in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.style_settings_to_website_theme in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.rename_ref_type_fieldnames in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.fix_email_alert in site1.local (1bd3e0294d)
Success
Executing frappe.patches.v5_0.fix_null_date_datetime in site1.local (1bd3e0294d)
Success
Executing execute:frappe.permissions.reset_perms("GL Entry") #2014-06-09 in site1.local (1bd3e0294d)
Success
Executing execute:frappe.permissions.reset_perms("Stock Ledger Entry") #2014-06-09 in site1.local (1bd3e0294d)
Success
Executing execute:frappe.rename_doc("DocType", "Support Ticket", "Issue", force=True) in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v4_4.make_email_accounts in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v4_2.reset_bom_costs in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.update_frozen_accounts_permission_role in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.update_dn_against_doc_fields in site1.local (1bd3e0294d)
Success
Executing execute:frappe.db.sql("update `tabMaterial Request` set material_request_type = 'Material Transfer' where material_request_type = 'Transfer'") in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('stock', 'doctype', 'item') in site1.local (1bd3e0294d)
Success
Executing execute:frappe.db.sql("update `tabItem` i set apply_warehouse_wise_reorder_level=1, re_order_level=0, re_order_qty=0 where exists(select name from `tabItem Reorder` where parent=i.name)") in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.set_default_company_in_bom in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.capacity_planning in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('crm', 'doctype', 'lead') in site1.local (1bd3e0294d)
Success
Executing execute:frappe.reload_doc('crm', 'doctype', 'opportunity') in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.rename_table_fieldnames in site1.local (1bd3e0294d)
Success
Executing execute:frappe.db.sql("update `tabJournal Entry` set voucher_type='Journal Entry' where ifnull(voucher_type, '')=''") in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v4_2.party_model in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.party_model_patch_fix in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.recalculate_total_amount_in_jv in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.update_companywise_payment_account in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.remove_birthday_events in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.update_item_name_in_bom in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.rename_customer_issue in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.rename_total_fields in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.new_crm_module in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.update_material_transfer_for_manufacture in site1.local (1bd3e0294d)
Success
Executing erpnext.patches.v5_0.update_item_description_and_image in site1.local (1bd3e0294d)
/home/revant/frappe-bench/apps/erpnext/erpnext/selling/doctype/opportunity_item/opportunity_item.json missing
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/revant/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 77, in <module>
    main()
  File "/home/revant/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 14, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/home/revant/frappe-bench/apps/frappe/frappe/commands.py", line 28, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/revant/frappe-bench/apps/frappe/frappe/commands.py", line 202, in migrate
    frappe.modules.patch_handler.run_all()
  File "/home/revant/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 25, in run_all
    if not run_single(patchmodule = patch):
  File "/home/revant/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 52, in run_single
    return execute_patch(patchmodule, method, methodargs)
  File "/home/revant/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 68, in execute_patch
    frappe.get_attr(patchmodule.split()[0] + ".execute")()
  File "/home/revant/frappe-bench/apps/erpnext/erpnext/patches/v5_0/update_item_description_and_image.py", line 29, in execute
    .format("item" if dt=="BOM" else "item_code", dt), as_dict=1)
  File "/home/revant/frappe-bench/apps/frappe/frappe/database.py", line 140, in sql
    self._cursor.execute(query)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
_mysql_exceptions.OperationalError: (1054, "Unknown column 'image' in 'where clause'")
Traceback (most recent call last):
  File "/usr/local/bin/bench", line 9, in <module>
    load_entry_point('bench==0.1', 'console_scripts', 'bench')()
  File "/home/revant/bench-repo/bench/cli.py", line 55, in cli
    bench()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 610, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 590, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 936, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 782, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/home/revant/bench-repo/bench/cli.py", line 247, in update
    patch_sites()
  File "/home/revant/bench-repo/bench/utils.py", line 118, in patch_sites
    run_frappe_cmd('--site', 'all', 'migrate', bench=bench)
  File "/home/revant/bench-repo/bench/utils.py", line 427, in run_frappe_cmd
    subprocess.check_call((f, '-m', 'frappe.utils.bench_helper', 'frappe') + args, cwd=sites_dir)
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('/home/revant/frappe-bench/env/bin/python', '-m', 'frappe.utils.bench_helper', 'frappe', '--site', 'all', 'migrate')' returned non-zero exit status 1
```

**Comment by pdvyas** [comment]
Fixed this. Can you `bench retry-upgrade`
Nice and detailed bug report, Thanks!

**Comment by revant** [comment]
Migration from fresh v4 site to v5 site was successful

**Comment by revant** [comment]
Error in same patch during migrating fresh dump of production on testing machine:

```
revant@revant-laptop:~/frappe-bench$ bench switch-to-develop
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.
Switched to branch 'develop'
Your branch is up-to-date with 'upstream/develop'.

Switched to develop
Please run `bench update --patch` to be safe from any differences in database schema
revant@revant-laptop:~/frappe-bench$ bench update --upgrade
Already up-to-date.
From https://github.com/frappe/shopping-cart
 * branch            develop    -> FETCH_HEAD
Already up-to-date.
From https://github.com/frappe/frappe
 * branch            develop    -> FETCH_HEAD
Already up-to-date.
From https://github.com/frappe/erpnext
 * branch            develop    -> FETCH_HEAD
Already up-to-date.
Migrating castlecraft.in
Executing frappe.patches.v5_0.convert_to_barracuda_and_utf8mb4 in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('core', 'doctype', 'docfield', force=True) #2015-01-01 in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('email', 'doctype', 'email_alert') #2014-07-15 in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('desk', 'doctype', 'todo') #2014-12-31-1 in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('custom', 'doctype', 'property_setter') #2014-12-31-1 in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.v4_to_v5 in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.remove_shopping_cart_app in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.permissions.reset_perms("Module Def") in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.permissions.reset_perms("User") #2015-03-24 in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.rename_table_fieldnames in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.communication_parent in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.clear_website_group_and_notifications in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.db.sql("""update tabComment set comment = substr(comment, 6, locate(":", comment)-6) where comment_type in ("Assigned", "Assignment Completed")""") in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.fix_feed in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.update_shared in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.bookmarks_to_stars in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.style_settings_to_website_theme in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.rename_ref_type_fieldnames in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.fix_email_alert in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.fix_null_date_datetime in castlecraft.in (cpes-frappe)
Success
Executing frappe.patches.v5_0.force_sync_website in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.permissions.reset_perms("GL Entry") #2014-06-09 in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.permissions.reset_perms("Stock Ledger Entry") #2014-06-09 in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.rename_doc("DocType", "Support Ticket", "Issue", force=True) in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v4_4.make_email_accounts in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v4_2.reset_bom_costs in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.update_frozen_accounts_permission_role in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.update_dn_against_doc_fields in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.db.sql("update `tabMaterial Request` set material_request_type = 'Material Transfer' where material_request_type = 'Transfer'") in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('stock', 'doctype', 'item') in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.db.sql("update `tabItem` i set apply_warehouse_wise_reorder_level=1, re_order_level=0, re_order_qty=0 where exists(select name from `tabItem Reorder` where parent=i.name)") in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.set_default_company_in_bom in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.capacity_planning in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('crm', 'doctype', 'lead') in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('crm', 'doctype', 'opportunity') in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.rename_table_fieldnames in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.db.sql("update `tabJournal Entry` set voucher_type='Journal Entry' where ifnull(voucher_type, '')=''") in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v4_2.party_model in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.party_model_patch_fix in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.recalculate_total_amount_in_jv in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.update_companywise_payment_account in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.remove_birthday_events in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.update_item_name_in_bom in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.rename_customer_issue in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.rename_total_fields in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.new_crm_module in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.update_material_transfer_for_manufacture in castlecraft.in (cpes-frappe)
Success
Executing execute:frappe.reload_doc('crm', 'doctype', 'opportunity_item') in castlecraft.in (cpes-frappe)
Success
Executing erpnext.patches.v5_0.update_item_description_and_image in castlecraft.in (cpes-frappe)
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/revant/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 77, in <module>
    main()
  File "/home/revant/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 14, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/revant/frappe-bench/env/local/lib/python2.7/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/home/revant/frappe-bench/apps/frappe/frappe/commands.py", line 28, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/revant/frappe-bench/apps/frappe/frappe/commands.py", line 202, in migrate
    frappe.modules.patch_handler.run_all()
  File "/home/revant/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 25, in run_all
    if not run_single(patchmodule = patch):
  File "/home/revant/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 52, in run_single
    return execute_patch(patchmodule, method, methodargs)
  File "/home/revant/frappe-bench/apps/frappe/frappe/modules/patch_handler.py", line 68, in execute_patch
    frappe.get_attr(patchmodule.split()[0] + ".execute")()
  File "/home/revant/frappe-bench/apps/erpnext/erpnext/patches/v5_0/update_item_description_and_image.py", line 33, in execute
    if cstr(d.description) == item_details.get(d.item_code).old_description:
AttributeError: 'NoneType' object has no attribute 'old_description'
Traceback (most recent call last):
  File "/usr/local/bin/bench", line 9, in <module>
    load_entry_point('bench==0.1', 'console_scripts', 'bench')()
  File "/home/revant/bench-repo/bench/cli.py", line 55, in cli
    bench()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 610, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 590, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 936, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 782, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/home/revant/bench-repo/bench/cli.py", line 247, in update
    patch_sites()
  File "/home/revant/bench-repo/bench/utils.py", line 118, in patch_sites
    run_frappe_cmd('--site', 'all', 'migrate', bench=bench)
  File "/home/revant/bench-repo/bench/utils.py", line 427, in run_frappe_cmd
    subprocess.check_call((f, '-m', 'frappe.utils.bench_helper', 'frappe') + args, cwd=sites_dir)
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('/home/revant/frappe-bench/env/bin/python', '-m', 'frappe.utils.bench_helper', 'frappe', '--site', 'all', 'migrate')' returned non-zero exit status 1
```

**Comment by rmehta** [comment]
@revant seems like you have items you may have deleted via a database query. Did you do that?

**Comment by revant** [comment]
I have imported lot of items with Data import tool and manipulated them through the frappe/erpnext interface only. 
I've not played with database on production ever. My production server is not in developer mode.

Everytime I try to test upgrade I take fresh copy of database from production on upgrade testing machine.

I played with testing db just to see what happens, then as you said I just take fresh/clean copy of db and test upgrade
